suite/tests/runcmp.cmake

```diff
@@ -57,32 +57,60 @@ if ("${capture}" STREQUAL "stderr")
   set(output "${cmd_err}")
 endif ()
 
-set(tmp "${cmp}-out")
-file(WRITE "${tmp}" "${output}")
+# Get expected output, we assume it has already been processed w/
+# regex => literal, etc.
+file(READ "${cmp}" expect)
 
-# We do not support regex in expect file b/c ctest can't handle big regex:
-#   "RegularExpression::compile(): Expression too big."
+# Convert file contents into a CMake list, where each element in the list is one
+# line of the file, empty lines are ignored.
+string(REGEX REPLACE "^[\n]+" "" output "${output}")
+string(REGEX REPLACE "[\n]+$" "" output "${output}")
+string(REPLACE "\n\n" "\n" output "${output}")
+STRING(REGEX REPLACE ";" "\\\\;" output "${output}")
+STRING(REGEX REPLACE "\n" ";" output "${output}")
```

Comment:
This looks much more complex than the approach I pointed you at in CMake_symbol_check.cmake:
```
string(REGEX MATCHALL "([^\n]+)\n" lines "${oneword}")
```
OK, most of the work here is about blank lines.  Do we really need to spend all this effort on leading and trailing and double newlines?  It seems ok to require the expect file to match the output file's newlines exactly: weren't we already doing that?